### PR TITLE
test(Redis Node): Continue on Fail for Redis (no-changelog)

### PR DESCRIPTION
**Pull request by dana-gill**
## Summary

This PR add tests for continue on fail for Redis. The PR resets mocks instead of clearing them to have a more clean state for testing in Redis.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-2147/redis-node-add-tests-and-investigate-success-also-outputting-on-error
Follow up for https://github.com/n8n-io/n8n/pull/11714#pullrequestreview-2492683245

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 1]
```diff
@@ -119,6 +132,17 @@ master_failover_state:no-failover
 					master_failover_state: 'no-failover',
 				});
 			});
+
+			it('should return an error on continue on fail', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:136]:
```suggestion
			it('should return an error when continue on fail is enabled', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 2]
```diff
@@ -132,6 +156,20 @@ master_failover_state:no-failover
 				expect(mockClient.del).toHaveBeenCalledWith('key1');
 				expect(output[0][0].json).toEqual({ x: 1 });
 			});
+
+			it('should return an error on continue on fail', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:160]:
```suggestion
			it('should return an error when continue on fail is enabled', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 3]
```diff
@@ -172,6 +210,19 @@ master_failover_state:no-failover
 					},
 				});
 			});
+
+			it('should return an error on continue on fail', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:214]:
```suggestion
			it('should return an error when continue on fail is enabled', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 4]
```diff
@@ -200,6 +251,19 @@ master_failover_state:no-failover
 				expect(mockClient.keys).toHaveBeenCalledWith('key*');
 				expect(output[0][0].json).toEqual({ key1: 'value1', key2: 'value2' });
 			});
+
+			it('should return an error on continue on fail', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:255]:
```suggestion
			it('should return an error when continue on fail is enabled', async () => {
```

**Review by ShireenMissi — COMMENTED** [review]
Mostly looks good 🎉 left some minor comments to improve clarity 
also could you add tests for when continue on fail is set to false 🙏

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 5]
```diff
@@ -119,6 +132,28 @@ master_failover_state:no-failover
 					master_failover_state: 'no-failover',
 				});
 			});
+
+			it('should return an error when continue on fail is enabled', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:136]:
```suggestion
			it('should continue and return an error when continue on fail is enabled and an error is thrown', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 6]
```diff
@@ -119,6 +132,28 @@ master_failover_state:no-failover
...
+				thisArg.continueOnFail.mockReturnValue(true);
+				mockClient.info.mockRejectedValue(new Error('Redis error'));
+
+				const output = await node.execute.call(thisArg);
+
+				expect(mockClient.info).toHaveBeenCalled();
+				expect(output[0][0].json).toEqual({ error: 'Redis error' });
+			});
+
+			it('should throw an error on continue on fail', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:147]:
```suggestion
			it('should throw an error when continue on fail is disabled and an error is thrown', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 7]
```diff
@@ -132,6 +167,33 @@ master_failover_state:no-failover
...
+
+				mockClient.del.mockRejectedValue(new Error('Redis error'));
+
+				const output = await node.execute.call(thisArg);
+
+				expect(mockClient.del).toHaveBeenCalled();
+				expect(output[0][0].json).toEqual({ error: 'Redis error' });
+			});
+
+			it('should throw an error on continue on fail', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:185]:
```suggestion
			it('should throw an error when continue on fail is disabled and an error is thrown', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 8]
```diff
@@ -172,6 +234,31 @@ master_failover_state:no-failover
...
+				mockClient.type.calledWith('key1').mockResolvedValue('string');
+				mockClient.get.mockRejectedValue(new Error('Redis error'));
+
+				const output = await node.execute.call(thisArg);
+				expect(mockClient.get).toHaveBeenCalled();
+
+				expect(output[0][0].json).toEqual({ error: 'Redis error' });
+			});
+
+			it('should throw an error when continue on fail is enabled', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:251]:
```suggestion
			it('should throw an error when continue on fail is disabled and an error is thrown', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 9]
```diff
@@ -132,6 +167,33 @@ master_failover_state:no-failover
 				expect(mockClient.del).toHaveBeenCalledWith('key1');
 				expect(output[0][0].json).toEqual({ x: 1 });
 			});
+
+			it('should return an error when continue on fail is enabled', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:171]:
```suggestion
			it('should continue and return an error when continue on fail is enabled and an error is thrown', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 10]
```diff
@@ -172,6 +234,31 @@ master_failover_state:no-failover
 					},
 				});
 			});
+
+			it('should return an error when continue on fail is enabled', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:238]:
```suggestion
			it('should continue and return an error when continue on fail is enabled and an error is thrown', async () => {
```

File: packages/nodes-base/nodes/Redis/test/Redis.node.test.ts [hunk 11]
```diff
@@ -200,6 +287,31 @@ master_failover_state:no-failover
 				expect(mockClient.keys).toHaveBeenCalledWith('key*');
 				expect(output[0][0].json).toEqual({ key1: 'value1', key2: 'value2' });
 			});
+
+			it('should return an error when continue on fail is enabled', async () => {
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Redis/test/Redis.node.test.ts:291]:
```suggestion
			it('should continue and return an error when continue on fail is enabled and an error is thrown', async () => {
```

**Review by ShireenMissi — COMMENTED** [review]
left some comments to improve clarity apart from that looks good 👍

**Review by ShireenMissi — APPROVED** [review]
LGTM 🎉

**Comment by janober** [comment]
Got released with [`n8n@1.73.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.73.0)

**Pull request merged**